### PR TITLE
Gracefully handle invalid cache metadata

### LINKED ISSUE #1094: Prism Launcher no longer starting properly
### Operating System

Windows

### Version of Prism Launcher

Prism Launcher 6.3

### Version of Qt

Unknown

### Description of bug

When launching Prism Launcher (I'm using it in portable mode) nothing will appear and the task will appear then disappear in task manager. The only change seems to be that PrismLauncher-0.log updates with the latest attempt. This always ends with the last couple lines being:
<>Accounts loaded.
Exception: "HttpMetaCache root is not an object"

Below is a full post of the log file create by Prism Launcher when trying to launch.

```
     0.000 D | <> Log initialized.
     0.000 D | <> No migration needed from "PolyMC"
     0.000 D | <> No migration needed from "MultiMC"
     0.000 D | "Prism Launcher" , (c) 2013-2021  "Prism Launcher Contributors\n© 2021-2022 PolyMC Contributors \n© 2012-2021 MultiMC Contributors"
     0.000 D | Version                    :  "6.3"
     0.000 D | Git commit                 :  "39bba9cbfadc0aac3ef571a80e78a9fa3f5a0028"
     0.000 D | Git refspec                :  "refs/heads/stable"
     0.000 D | Work dir before adjustment :  "D:/PrismLauncher"
     0.000 D | Work dir after adjustment  :  "D:/PrismLauncher"
     0.000 D | Adjusted by                :  "Portable data path"
     0.000 D | Binary path                :  "D:/PrismLauncher"
     0.000 D | Application root path      :  "D:/PrismLauncher"
     0.000 D | <> Paths set.
     0.003 D | Detected default console font: "Courier New" , substitutions: ""
     0.007 D | <> Settings loaded.
     0.007 D | Detecting proxy settings...
     0.007 D | Using no proxy is an option!
     0.007 D | <> Network done.
     0.008 C | Exception: "Document: Error parsing JSON: illegal value"
     0.008 C | Translations Download Failed: index file could not be parsed as json
     0.008 W | Selected invalid language "" , defaulting to "en_US"
     0.008 D | Your language is ""
     0.008 D | <> Translations loaded.
     0.009 D | Started watching  "D:/PrismLauncher/icons"
     0.009 D | Adding  "icons/curseforge_638137997057123045"
     0.099 D | Adding  "icons/curseforge_638104922297531668"
     0.101 D | Adding  "icons/curseforge_637967192029411805"
     0.103 D | Sorting icon list...
     0.103 D | <> Instance icons intialized.
     0.103 D | [Theme] <> Icon themes initialized.
     0.103 D | [Theme] <> Initializing Widget Themes
     0.103 D | [Theme] Determining System Theme...
     0.103 D | [Theme] System theme seems to be: "windowsvista"
     0.103 D | [Theme] Considering theme from theme factory: "windowsvista"
     0.103 D | [Theme] System theme has been determined to be: "windowsvista"
     0.103 D | [Theme] Loading Built-in Theme: "system"
     0.103 D | [Theme] Loading Built-in Theme: "dark"
     0.103 D | [Theme] Loading Built-in Theme: "bright"
     0.103 D | [Theme] Theme Folder Path:  "D:/PrismLauncher/themes"
     0.103 D | [Theme] <> Widget themes initialized.
     0.103 D | Instance path              :  "instances"
     0.103 D | Loading Instances...
     0.103 D | Discovering instances in "D:/PrismLauncher/instances"
     0.103 D | Found instance ID "1.19.2"
     0.103 D | Found instance ID "GT_New_Horizons_2.3.3_Java_17-20"
     0.103 D | Found instance ID "Medieval MC [FORGE] - MMC3"
     0.103 D | Found instance ID "Custom 0.0.1"
     0.104 D | Found instance ID "Create- Astral"
     0.104 D | Found instance ID "Custom-0.0.1"
     0.104 D | Will load group list now.
     0.104 D | Group list loaded.
     0.104 D | Loaded instance  "1.19.2"  from  "D:/PrismLauncher/instances/1.19.2"
     0.105 D | Loaded instance  "GT_New_Horizons_2.3.3_Java_17-20"  from  "D:/PrismLauncher/instances/GT_New_Horizons_2.3.3_Java_17-20"
     0.105 D | Loaded instance  "Medieval MC [FORGE] - MMC3"  from  "D:/PrismLauncher/instances/Medieval MC [FORGE] - MMC3"
     0.106 D | Loaded instance  "Custom 0.0.1"  from  "D:/PrismLauncher/instances/Custom 0.0.1"
     0.106 D | Loaded instance  "Create: Astral"  from  "D:/PrismLauncher/instances/Create- Astral"
     0.107 D | Loaded instance  "Zeitlich"  from  "D:/PrismLauncher/instances/Custom-0.0.1"
     0.107 D | <> Instances loaded.
     0.107 D | Loading accounts...
     0.107 C | "Failed to read the account list file (accounts.json)."
     0.107 D | <> Accounts loaded.
     0.107 C | Exception: "HttpMetaCache root is not an object"
```

### Steps to reproduce

1: Double click prismlauncher.exe
2: Fail.

### Suspected cause

Something with the exception at the end of the log I assume since that's where it seems to just stop dead.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixed #1094

### CODE REVIEW

File: launcher/net/HttpMetaCache.cpp [hunk 1]
```diff
@@ -218,9 +218,24 @@ void HttpMetaCache::Load()
...
+                           .arg(parseError.errorString(), QString::number(parseError.offset))
+                           .toUtf8();
+        return;
+    }
+
+    // Make sure the root is an object.
+    if (!json.isObject()) {
+        qCritical() << "HttpMetaCache root should be an object.";
+        return;
+    }
```
Comment (Scrumplex) [inline launcher/net/HttpMetaCache.cpp:236]: maybe just start from scratch if the file is invalid. It doesn't store important data anyway so might as well just ignore it and start with a clean state

Comment (Trial97) [reply launcher/net/HttpMetaCache.cpp:236]: that is what my code does it just prints the logs and exits the load function similar to how it does when it fails to open it.

Comment (Trial97) [reply launcher/net/HttpMetaCache.cpp:236]: Do you want to drop the logs?

Comment (Scrumplex) [reply launcher/net/HttpMetaCache.cpp:236]: I wasn't aware of the context we are in :sweat_smile:

Code is fine then! :D

**Review by Scrumplex — APPROVED** [review]

**Review by TayouVR — APPROVED** [review]

**Comment by TheKodeToad** [comment]
Why is toUtf8 needed?

**Comment by Trial97** [comment]
copied from another place please make a suggestion if you want it removed

**Comment by TheKodeToad** [comment]
There was probably a Good Reason then

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**